### DISCUSSION TITLE
Is Genesis also using contact detection and MLCP iterative solving?

### DISCUSSION THREAD

**Discussion by oneanime** [Genesis-Embodied-AI/genesis-world#206] [Q&A] [answered]
Is Genesis also using contact detection and MLCP iterative solving? Why is it reportedly 80 times faster than PhysX in computations?

**Comment by zhouxian** [comment]
@YilingQiao could you answer this?

**Comment by YilingQiao** [comment, marked as answer]
We are not using MLCP but are instead following MuJoCo's convex optimization formula. Our collision detection utilizes a precomputed SDF-style support function for acceleration. Additionally, the warm start in collision resolving can save a lot of time for the solver.
